src/finch/util/config.py

```diff
@@ -0,0 +1,68 @@
+import os
+import json
+
+"""
+Finch Configuration Module
+
+This module manages configuration settings for the Finch application. 
+Finch stores its settings and data in the `FINCH_PATH` directory, which 
+defaults to `~/.finch` but can be customized using the `FINCH_PATH` 
+environment variable.
+
+Configuration details:
+- Settings are stored in a `config.json` file within the `FINCH_PATH` directory.
+- Values can be set via environment variables, the `config.json` file, 
+    or the `set_config` function.
+- Configuration values are loaded automatically when the module is imported 
+    and can be accessed using the `get_config` function.
+
+Use this module to easily manage and retrieve Finch-specific settings.
+"""
+
+depot_dir = os.path.realpath(os.path.expanduser(os.getenv('FINCH_PATH', os.path.join("~", ".finch"))))
+
+default_config = {
+    "FINCH_CACHE_PATH": os.path.join(depot_dir, "cache"),
+    "FINCH_CACHE_SIZE": 10000,
+    "FINCH_CACHE_ENABLE": True,
+    "FINCH_TMP": os.path.join(depot_dir, "tmp"),
+    "FINCH_LOG_PATH": os.path.join(depot_dir, "log.txt"),
+    "FINCH_CC": "gcc",
```

Comment:
I'd use `os.environ.get("CC")`. If we intend to compile C++, I'd change this to `FINCH_CXX` and use `os.environ.get("CXX")`.

Reply:
I think we're doing C for now, I'll use that variable, great catch!

Reply:
I'm remembering my worry with CC was that sometimes compiler flags don't translate between compilers, so I thought I'd push the user to read the config if they don't want to use gcc and figure out if the flags are okay. That said, I think `-fPIC -O3` works on all linux systems. For windows, we don't need -fpic, and things are different.